### PR TITLE
canadian provincial data API/GUI added

### PR DESCRIPTION
**Pull request by DexterHarvey**

### CODE REVIEW

File: DataProviders/CanadaProvCovid19DataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class CanadaProvCovid19DataProvider {
+        private config = {
+        baseURL: 'https://services9.arcgis.com/pJENMVYPQqZZe20v/arcgis/rest/services/Join_Features_to_Enriched_Population_Case_Data_By_Province_Polygon/FeatureServer/0',
+        timeout: 1000,
+        headers: {
+            'content-type': 'JSON',
+        }
+    };
+    public async getDataForProv(provTerritory: String) {
+        // const result = await axios.get(`query?where=NAME%20%3D%20\'${provTerritory}\'&outFields=OBJECTID,NAME,Recovered,Tests,Last_Updated,SourceURL,ActiveCases,Hospitalized,ICU,ICUVent,Deaths,Case_Total,GlobalID_2,Vaccinated,SecondURL,ThirdURL,Abbreviation&returnGeometry=false&outSR=&f=json`, this.config);
```
Comment (jeffroutledge) [inline DataProviders/CanadaProvCovid19DataProvider.ts]: Remove commented code.

File: Models/TerritoryStats.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    LastUpdated: number = 0;
+    SourceURL: string = '';
+    ActiveCases: number = 0;
+    Hospitalized: number = 0;
+    ICU: number = 0;
+    ICUVentilator: number = 0;
+    Deaths: number = 0;
+    CaseTotal: number = 0;
+    Vaccinated: number = 0;
+    constructor (objectId: number, name: string, recovered: number, tests: number, lastUpdated: number, sourceURL: string, activeCases: number, hospitalized: number, iCU: number, iCUVentilator: number, deaths: number, caseTotal: number, vaccinated: number) {
```
Comment (jeffroutledge) [inline Models/TerritoryStats.ts:15]: These properties should be public and not set in the constructor.

File: Models/TerritoryStats.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+class TerritoryStats {
```
Comment (jeffroutledge) [inline Models/TerritoryStats.ts:1]: This should be an interface.

File: client/src/TerritoryStats.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+class TerritoryStats {
```
Comment (jeffroutledge) [inline client/src/TerritoryStats.ts]: Ideally the frontend would use the same model as the backend so update this to match or use the backend model directly for now.

**Review by jeffroutledge — COMMENTED** [review]
A couple things to clean up.

File: Models/TerritoryStats.ts (hunk 3 above)
Comment (jeffroutledge) [reply Models/TerritoryStats.ts:1]: https://blog.logrocket.com/interfaces-in-typescript-what-are-they-and-how-do-we-use-them-befbc69b38b3/

File: DataProviders/CanadaProvCovid19DataProvider.ts (hunk 1 above)
Comment (DexterHarvey) [reply DataProviders/CanadaProvCovid19DataProvider.ts]: done

File: Models/TerritoryStats.ts (hunk 2 above)
Comment (DexterHarvey) [reply Models/TerritoryStats.ts:15]: done

File: Models/TerritoryStats.ts (hunk 3 above)
Comment (DexterHarvey) [reply Models/TerritoryStats.ts:1]: done

File: client/src/TerritoryStats.ts (hunk 4 above)
Comment (DexterHarvey) [reply client/src/TerritoryStats.ts]: React does not allow the use of files outside (above?) the src/ folder if initialized as a create-react-app due to config. This is possible to correct either with a library such as lerna, rewire, or by ejecting the project and adjusting the webpack config manually. This seems above my pay grade, so I'll let you try to handle it without exploding everything https://stackoverflow.com/questions/44114436/the-create-react-app-imports-restriction-outside-of-src-directory

**Pull request merged**